core/src/main/scala/kafka/server/OffsetCheckpoint.scala

```diff
@@ -50,14 +52,20 @@ class OffsetCheckpoint(val file: File) extends Logging {
           writer.write("%s %d %d".format(topicPart.topic, topicPart.partition, offset))
           writer.newLine()
         }
-      
+
         // flush the buffer and then fsync the underlying file
         writer.flush()
         fileOutputStream.getFD().sync()
+      } catch {
+        case e: FileNotFoundException =>
+          if (FileSystems.getDefault.isReadOnly) {
+            fatal("Halting writes to offset checkpoint file because the underlying file system is inaccessible : ", e)
+            Runtime.getRuntime.halt(1)
+          }
```

Comment:
@MayureshGharat, by conflicts I mean that it doesn't merge cleanly to master anymore (you need to merge master into your branch).

I am OK with filing a separate JIRA for tackling the bigger issue of how we do the halting. Let's see what the committers think.